packages/frontend/src/features/sqlRunner/hooks/useSqlQueryRun.tsx

```diff
@@ -50,169 +47,73 @@ export const useSqlQueryRun = ({
-                throw new Error('Missing file URL');
-            }
-            const response = await fetch(url, {
-                method: 'GET',
-                headers: {
-                    Accept: 'application/json',
-                },
+        async ({ sql, limit }) => {
+            const scheduledJob = await scheduleSqlJob({
+                projectUuid,
+                sql,
+                limit,
             });
-            const rb = response.body;
-            const reader = rb?.getReader();
-
-            const stream = new ReadableStream({
-                start(controller) {
-                    function push() {
-                        void reader?.read().then(({ done, value }) => {
-                            if (done) {
-                                // Close the stream
-                                controller.close();
-                                return;
-                            }
-                            // Enqueue the next data chunk into our target stream
-                            controller.enqueue(value);
 
-                            push();
-                        });
-                    }
+            const job = await getSqlRunnerCompleteJob(scheduledJob.jobId);
 
-                    push();
-                },
-            });
-
-            const responseStream = new Response(stream, {
-                headers: { 'Content-Type': 'application/json' },
-            });
-            const result = await responseStream.text();
+            if (job.status === SchedulerJobStatus.ERROR) {
+                if (isErrorDetails(job.details)) {
+                    showToastError({
+                        title: 'Could not run SQL query',
+                        subtitle: job.details.error,
+                    });
+                }
+                return undefined;
+            }
 
-            // Split the JSON strings by newline
-            const jsonStrings = result.trim().split('\n');
-            const jsonObjects: ResultRow[] = jsonStrings
-                .map((jsonString) => {
-                    try {
-                        return JSON.parse(jsonString);
-                    } catch (e) {
-                        throw new Error('Error parsing JSON');
-                    }
-                })
-                .filter((obj) => obj !== null);
+            const url =
+                job.details && !isErrorDetails(job.details)
+                    ? job.details.fileUrl
+                    : undefined;
+            const results = await getResultsFromStream(url);
 
-            return jsonObjects;
+            return {
+                results,
+                columns:
+                    job.details && !isErrorDetails(job.details)
+                        ? job.details.columns
+                        : [],
+            };
         },
         {
+            mutationKey: ['sqlRunner', 'run'],
             onError: () => {
                 showToastError({
```

Comment:
do we want to clear previous data on error? 


In this PR:  
1. User types query with results
2. User types a query that leads to no results
3. Previous data is there, but an error toast message appears

Should we clear the previous data and have an empty state?
Perhaps it's a nice experience to keep the previous data, but unsure!